mlrun/model.py

```diff
@@ -169,7 +169,12 @@ def from_dict(cls, classes_map: dict, children=None, default_kind=""):
 
         new_obj = cls(classes_map, default_kind)
         for name, child in children.items():
-            child_obj = new_obj._get_child_object(child, name)
+            obj_name = name
+            if hasattr(child, "name") and child.name is not None:
+                obj_name = child.name
+            elif isinstance(child, dict) and "name" in child:
+                obj_name = child["name"]
```

Comment:
Is this fixing some issue? A bug? Looks like you're overriding the name in the `children` list if the object itself has a `name` - why is this necessary?